return scale loss

Ghostvv opened the pull request:
**Proposed changes**:
- fixes https://github.com/RasaHQ/rasa/issues/5451
- add loss scaling to entity loss
- do not scale mask loss
- only scale if some of the examples in a batch has probability of the golden label more than `0.5`

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

tabergma commented on rasa/nlu/classifiers/diet_classifier.py:
```diff
@@ -1139,20 +1139,22 @@ def _prepare_mask_lm_layers(self, name: Text) -> None:
         self._prepare_embed_layers(f"{name}_golden_token")
 
-        self._prepare_dot_product_loss(f"{name}_mask")
+        self._prepare_dot_product_loss(f"{name}_mask", scale_loss=False)
```
Should we add a comment here why we always set `scale_loss` to `False`?

tabergma commented:
Also, I think we should add a changelog entry.

Ghostvv commented:
@tabergma I updated changelog and changed the logic of scaling loss: now it only turns on, if at least 1 example in a batch is learned (making training loss curve jump to higher values after some epoch)

don't know what to do with `DeepSource` check: I didn't change anything regarding the bugs, it raises

tabergma commented:
DeepSource checks the complete file you touched. If the checks are not critical, I guess you can ignore them for now. I think we are fine here. But ideally all of them are tackled.
